boot/seal.go

```diff
@@ -957,3 +957,14 @@ func postFactoryResetCleanup() error {
 
 	return nil
 }
+
+// resealExpectedByModeenvChange returns true if resealing is expected
+// due to modeenv changes, false otherwise. Reseal might not be needed
+// if the only change in modeenv is the gadget (if the boot assets
+// change that is detected in resealKeyToModeenv() and reseal will
+// happen anyway)
+func resealExpectedByModeenvChange(old *Modeenv, new *Modeenv) bool {
```

Comment:
looking at how it is used old and new should be swapped? but given the definition it's not really relevant so maybe:

m1, m2 *Modeenv

(golang let's specify repeated types once)

Also does this merit its own short unit tests (via export_test.go exposing) ?